services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -132,31 +169,78 @@ public int hashCode() {
+         * and constraints.
+         *
+         * @param queryConditional the query conditions
+         * @return a builder of this type
+         */
         public Builder queryConditional(QueryConditional queryConditional) {
             this.queryConditional = queryConditional;
             return this;
         }
 
+        /**
+         * Results are sorted by sort key in ascending order if {@link #scanIndexForward} is true. If its false, the
+         * order is descending. The default value is true.
+         *
+         * @param scanIndexForward the sort order
+         * @return a builder of this type
+         */
         public Builder scanIndexForward(Boolean scanIndexForward) {
             this.scanIndexForward = scanIndexForward;
             return this;
         }
 
+        /**
+         * The primary key of the first item that this operation will evaluate. Use the value that was returned for
+         * {@link Page#lastEvaluatedKey()} in the previous operation.
+         *
+         * @param exclusiveStartKey the primary key value to start evaluate
+         * @return a builder of this type
+         */
         public Builder exclusiveStartKey(Map<String, AttributeValue> exclusiveStartKey) {
             this.exclusiveStartKey = exclusiveStartKey != null ? new HashMap<>(exclusiveStartKey) : null;
             return this;
         }
 
+        /**
+         * Sets a limit on how many items to evaluate in the query. If not set, the operation uses
+         * the maximum values allowed.
+         * <p/>
+         * <b>Note:</b>The limit does not refer to the number of items to return, but how many items
+         * the database should evaluate while executing the query. Use limit together with {@link Page#lastEvaluatedKey()}
+         * and {@link #exclusiveStartKey} in subsequent query calls to evaluate <em>limit</em> items per call.
+         *
+         * @param limit the maximum number of items to evalute
+         * @return a builder of this type
+         */
         public Builder limit(Integer limit) {
             this.limit = limit;
             return this;
         }
 
+        /**
+         Determines the read consistency model: If set to true, the operation uses strongly consistent reads; otherwise,
+         the operation uses eventually consistent reads.
+         *
+         * @param consistentRead sets consistency model of the operation to use strong consistency
+         * @return a builder of this type
+         */
         public Builder consistentRead(Boolean consistentRead) {
             this.consistentRead = consistentRead;
             return this;
         }
 
+        /**
+         * Refines the query results by applying the filter expression on the results returned
+         * from the query and discards items that do not match. See {@link Expression} for examples
+         * and constraints.
```

Comment:
We should make it clear that this filter does not reduce the cost of your query, it is applied after the database finds items matching your QueryConditional.

Reply:
Agree, added.